### PR TITLE
Bugfix/nw23001440/269 data reference

### PR DESCRIPTION
**Pull request by davidepalladino-apuliasoft**
## Description
This branch fixes two problem:
-  the nonexistent `Z` RPG type. In addition, the `ERROR06` test was fixed by considering that type is now avilable, changing to one that doesn't exist;
- the resolution of an inline Data Definition that must still be executed. In RPG is possible to use the `*LIKE` from a subroutine to a data definition of another subroutine.

## Checklist:
- [x] There are tests regarding this feature
- [x] The code follows the Kotlin conventions (run `./gradlew ktlintCheck`)
- [x] The code passes all tests (run `./gradlew check`)
- [ ] There is a specific documentation in the `docs` directory

### CODE REVIEW

**Review by lanarimarco — COMMENTED** [review]
I am aware that the implementation of DefineStmt is a mess because of its recursive behavior. I worked on it a few weeks ago, and it was really challenging. Your work, then, is very good. I just ask you for a little refactor here and there

File: rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/ast/statements.kt [hunk 1]
```diff
@@ -1181,6 +1198,12 @@ private fun DefineStmt.exitFromStack(): Boolean {
     return stack.remove(this)
 }
 
+private fun DefineStmt.getStack(): List<DefineStmt> {
```
[the remark this answers is not in the document]
Comment (davidepalladino-apuliasoft) [reply rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/ast/statements.kt]: Done

File: rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/ast/statements.kt [hunk 2]
```diff
@@ -1146,18 +1148,33 @@ data class DefineStmt(
     override fun execute(interpreter: InterpreterCore) {
         // Nothing to do here
     }
+
+    private fun findInGlobalScope(cu: CompilationUnit, search: String): InStatementDataDefinition? {
+        var inStatementDataDefinition = cu.main.stmts.findInStatementDataDefinition(originalName, this)
```
[the remark this answers is not in the document]
Comment (davidepalladino-apuliasoft) [reply rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/ast/statements.kt]: Done

**Review by lanarimarco — APPROVED** [review]
Ok

**Pull request merged**